components/board/genericlinux/board.go

```diff
@@ -466,20 +468,29 @@ func (b *Board) SetPowerMode(
 func (b *Board) StreamTicks(ctx context.Context, interrupts []board.DigitalInterrupt, ch chan board.Tick,
 	extra map[string]interface{},
 ) error {
+	var rawInterrupts []*digitalInterrupt
 	for _, i := range interrupts {
-		pinwrappers.AddCallback(i.(*pinwrappers.BasicDigitalInterrupt), ch)
+		raw, ok := i.(*digitalInterrupt)
+		if !ok {
+			return errors.New("cannot stream ticks to an interrupt not associated with this board")
```

Comment:
nit but maybe change `to` to `of`. Also if one of the interrupts is bad do we want to fail the whole streamticks call or should we still stream the valid interrupts?

Reply:
> if one of the interrupts is bad do we want to fail the whole streamticks call or should we still stream the valid interrupts?
Excellent question! I _think_ we should give up on the whole attempt, but could be convinced otherwise:
- for the ultrasonic sensor, if we can't get interrupts from the one pin, the entire sensor won't work, and we'll want to return errors to the user
- for quadrature encoders, even if one interrupt works, if the second one doesn't, we will be unable to record movement (the one interrupt will fire, resulting in us thinking we're repeatedly going backwards and forwards a quarter turn). I'd rather return an error to inform the user.
- for any other time when we're streaming from a single pin, if that pin fails, we won't stream anything, regardless of which approach we take.
- I can't think of a second use case for streaming multiple interrupts. If you can, let's think that through.